retrodep/gosource.go

```diff
@@ -543,6 +543,8 @@ func (src GoSource) RepoPathForImportPath(importPath string) (*RepoPath, error)
 		if nerr == nil {
 			return &RepoPath{RepoRoot: *r}, nil
 		}
+	} else if err != nil && r == nil {
+		return nil, err
```

Comment:
I don't feel we should be looking at `r` at all if `err != nil`. Also, we haven't fixed the second occurrence of the problem.

Would this be easier to read if we had the `err != nil` check only once?
```
if err != nil {
    // Special handling only if '_' is in the import path.
    if !strings.ContainsRune(importPath, '_') {
        return nil, err
    }

    // gopkg.in gives ...
    ...
    if nerr != nil {
        return nil, err  // original error
    }

    return &RepoPath{...}, nil
}
```

Reply:
done